### ISSUE TITLE
[studio] Deleting a page that has subpages shows misleading information about deleted items

### ISSUE THREAD

**Issue by grundmann-thalia** [craftercms/craftercms#4182]
## Describe the bug
Given a page that has three subpages - when I click "Delete" on the page the Delete-dialog tells me in the "Child Items"-section that the three subpages will get deleted but in fact they will be not deleted.

## To Reproduce
1. Create a page with three subpages
2. Delete page and confirm
3. Reload tree
4. The deleted page is now just a folder with three subpages

## Expected behavior
Delete-dialog should not tell that the subpages will be deleted

## Screenshots
Screenshot of Delete-Dialog:
![image](https://user-images.githubusercontent.com/58164705/87158841-a53d7880-c2c0-11ea-9967-e0994dd46917.png)
Screenshot of resulting tree after deletion:
![image](https://user-images.githubusercontent.com/58164705/87160068-a53e7800-c2c2-11ea-94e3-7d3394546a0d.png)

## Logs
n.a.

### Version
3.1.4
### OS
Linux
### Browser
Chrome

## Additional context
n.a.

**Comment by russdanner** [comment]
I have investigated this issue:
- Deletes are propagated all the way to publish as expected.
- On refresh you will see the folder where the page has been removed and the child objects will be displayed.
- A delete of the folder propagates to publish as expected.

Expected behavior:
The specified functionality requires that the page and all of the children are deleted (as shown by the dependencies.). If you do not want to delete the children you should first move them before performing the delete (similar to a file system.)

Fix:
For now, we need to fix the system to delete the top level folder rather than the xml

Enhancement for future release:
We can add an improvement tag to allow you to check/un-check a cascaded child delete for a later release.

**Comment by dejan-brkic** [comment]
Done
PR:
https://github.com/craftercms/studio-ui/pull/1873
https://github.com/craftercms/studio/pull/2116
https://github.com/craftercms/studio/pull/2117

**Comment by dejan-brkic** [comment]
@rart Please apply changes in develop as agreed

**Comment by sumerjabri** [comment]
@rart it's merged and ready for you to update the UI.

**Comment by rart** [comment]
@jvega190, this needs to be ported to 3.2; ping me to discuss.

**Comment by sumerjabri** [comment]
@yacdaniel please validate in 3.1 so we can close it for 3.1 and focus on 3.2.

**Comment by yacdaniel** [comment]
@dejan-brkic the reported issue still is not fixed, when the parent page is deleted the child content is deleted but the page becomes an empty folder when it should have been deleted

**Structure**
![image](https://user-images.githubusercontent.com/6722074/91770735-ec4c3980-eb9e-11ea-830e-66253edd22ec.png)

After right click `toppage` in sidebar and delete
![image](https://user-images.githubusercontent.com/6722074/91772192-9b8a1000-eba1-11ea-94cd-de00c9df7d9e.png)

Also, when there are folders under the parent being deleted, the folders are left on disk after deletion

**Structure**
![image](https://user-images.githubusercontent.com/6722074/91771232-df7c1580-eb9f-11ea-80fb-d785da5f90d0.png)

After righ click `topcontent` and delete it
![image](https://user-images.githubusercontent.com/6722074/91769681-11d84380-eb9d-11ea-9e8d-362bf2398616.png)

**Structure**
![image](https://user-images.githubusercontent.com/6722074/91773384-e86ee600-eba3-11ea-80aa-20e45b4c2244.png)

After righ click `page1` and delete it
![image](https://user-images.githubusercontent.com/6722074/91772754-c2951180-eba2-11ea-9341-1e5e007d2e1e.png)

**Comment by dejan-brkic** [comment]
Done
PR:
https://github.com/craftercms/studio/pull/2156
https://github.com/craftercms/studio/pull/2157

**Comment by yacdaniel** [comment]
@dejan-brkic there is a publisher error when a item is deleted
```
[INFO] 2020-09-03T08:48:48,587 [Studio_Publisher_Thread_Pool9] [repository.GitContentRepository] | Not able to find branch live for site empty123. Creating new branch 
[ERROR] 2020-09-03T08:48:48,595 [Studio_Publisher_Thread_Pool9] [repository.GitContentRepository] | Error when publishing site empty123 to environment live 
java.nio.file.NoSuchFileException: /home/yacdaniel/work/craftercms3.1.xE/crafter-authoring/data/repos/sites/empty123/published/site/website/topparent
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86) ~[?:1.8.0_252]
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102) ~[?:1.8.0_252]
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107) ~[?:1.8.0_252]
	at sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes(UnixFileAttributeViews.java:55) ~[?:1.8.0_252]
	at sun.nio.fs.UnixFileSystemProvider.readAttributes(UnixFileSystemProvider.java:144) ~[?:1.8.0_252]
	at sun.nio.fs.LinuxFileSystemProvider.readAttributes(LinuxFileSystemProvider.java:99) ~[?:1.8.0_252]
	at java.nio.file.Files.readAttributes(Files.java:1737) ~[?:1.8.0_252]
	at java.nio.file.FileTreeWalker.getAttributes(FileTreeWalker.java:219) ~[?:1.8.0_252]
	at java.nio.file.FileTreeWalker.visit(FileTreeWalker.java:276) ~[?:1.8.0_252]
	at java.nio.file.FileTreeWalker.walk(FileTreeWalker.java:322) ~[?:1.8.0_252]
	at java.nio.file.FileTreeIterator.<init>(FileTreeIterator.java:72) ~[?:1.8.0_252]
	at java.nio.file.Files.walk(Files.java:3574) ~[?:1.8.0_252]
	at org.craftercms.studio.impl.v2.repository.GitContentRepository.deleteParentFolder(GitContentRepository.java:1100) ~[classes/:3.1.10-SNAPSHOT]
	at org.craftercms.studio.impl.v2.repository.GitContentRepository.publish(GitContentRepository.java:985) [classes/:3.1.10-SNAPSHOT]
	at org.craftercms.studio.impl.v2.repository.blob.BlobAwareContentRepository.publish(BlobAwareContentRepository.java:513) [classes/:3.1.10-SNAPSHOT]
	at org.craftercms.studio.impl.v1.service.deployment.job.PublisherTask.deploy(PublisherTask.java:381) [classes/:3.1.10-SNAPSHOT]
	at org.craftercms.studio.impl.v1.service.deployment.job.PublisherTask.doPublishing(PublisherTask.java:251) [classes/:3.1.10-SNAPSHOT]
	at org.craftercms.studio.impl.v1.service.deployment.job.PublisherTask.run(PublisherTask.java:151) [classes/:3.1.10-SNAPSHOT]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_252]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_252]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_252]
[ERROR] 2020-09-03T08:48:48,596 [Studio_Publisher_Thread_Pool9] [job.PublisherTask] | Error while executing deployment to environment store for site "empty123", number of items "1" 
org.craftercms.studio.api.v1.service.deployment.DeploymentException: Error during deployment to environment live for site empty123
	at org.craftercms.studio.impl.v2.repository.blob.BlobAwareContentRepository.publish(BlobAwareContentRepository.java:515) ~[classes/:3.1.10-SNAPSHOT]
	at org.craftercms.studio.impl.v1.service.deployment.job.PublisherTask.deploy(PublisherTask.java:381) ~[classes/:3.1.10-SNAPSHOT]
	at org.craftercms.studio.impl.v1.service.deployment.job.PublisherTask.doPublishing(PublisherTask.java:251) [classes/:3.1.10-SNAPSHOT]
	at org.craftercms.studio.impl.v1.service.deployment.job.PublisherTask.run(PublisherTask.java:151) [classes/:3.1.10-SNAPSHOT]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_252]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_252]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_252]
Caused by: org.craftercms.studio.api.v1.service.deployment.DeploymentException: Error when publishing site empty123 to environment live [commit ID = 1ae5c66500e68b411acd7568bf7f207f9c06527c]
	at org.craftercms.studio.impl.v2.repository.GitContentRepository.publish(GitContentRepository.java:1077) ~[classes/:3.1.10-SNAPSHOT]
	at org.craftercms.studio.impl.v2.repository.blob.BlobAwareContentRepository.publish(BlobAwareContentRepository.java:513) ~[classes/:3.1.10-SNAPSHOT]
	... 6 more
[ERROR] 2020-09-03T08:48:48,601 [Studio_Publisher_Thread_Pool9] [job.PublisherTask] | Error while executing deployment to environment store for site: empty123 
org.craftercms.studio.api.v1.service.deployment.DeploymentException: Error during deployment to environment live for site empty123
	at org.craftercms.studio.impl.v2.repository.blob.BlobAwareContentRepository.publish(BlobAwareContentRepository.java:515) ~[classes/:3.1.10-SNAPSHOT]
	at org.craftercms.studio.impl.v1.service.deployment.job.PublisherTask.deploy(PublisherTask.java:381) ~[classes/:3.1.10-SNAPSHOT]
	at org.craftercms.studio.impl.v1.service.deployment.job.PublisherTask.doPublishing(PublisherTask.java:251) [classes/:3.1.10-SNAPSHOT]
	at org.craftercms.studio.impl.v1.service.deployment.job.PublisherTask.run(PublisherTask.java:151) [classes/:3.1.10-SNAPSHOT]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_252]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_252]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_252]
Caused by: org.craftercms.studio.api.v1.service.deployment.DeploymentException: Error when publishing site empty123 to environment live [commit ID = 1ae5c66500e68b411acd7568bf7f207f9c06527c]
	at org.craftercms.studio.impl.v2.repository.GitContentRepository.publish(GitContentRepository.java:1077) ~[classes/:3.1.10-SNAPSHOT]
	at org.craftercms.studio.impl.v2.repository.blob.BlobAwareContentRepository.publish(BlobAwareContentRepository.java:513) ~[classes/:3.1.10-SNAPSHOT]
	... 6 more
[INFO] 2020-09-03T08:48:48,604 [Studio_Publisher_Thread_Pool9] [job.PublisherTask] | Continue executing deployment for other sites. 
```

**Comment by dejan-brkic** [comment]
Done
PR:
https://github.com/craftercms/studio/pull/2159

**Comment by yacdaniel** [comment]
@dejan-brkic when the item to delete has child folders they are left empty on disk:
**Structure**
![image](https://user-images.githubusercontent.com/6722074/91773384-e86ee600-eba3-11ea-80aa-20e45b4c2244.png)

After righ click `page1` and delete it
![image](https://user-images.githubusercontent.com/6722074/91772754-c2951180-eba2-11ea-9341-1e5e007d2e1e.png)

**Comment by dejan-brkic** [comment]
Done
PR:
https://github.com/craftercms/studio/pull/2166
https://github.com/craftercms/studio/pull/2167

**Comment by yacdaniel** [comment]
Verified fixed in `3.1.10-SNAPSHOT`

**Comment by sumerjabri** [comment]
Test and validate for 3.2.0

**Comment by yacdaniel** [comment]
Verified fixed in `3.2.0`
